src/qt/bitcoin.cpp

```diff
@@ -251,6 +251,12 @@ BitcoinCore::BitcoinCore(interfaces::Node& node) :
 void BitcoinCore::handleRunawayException(const std::exception *e)
 {
     PrintExceptionContinue(e, "Runaway exception");
+    try {
+        m_node.appShutdown();
```

Comment:
This doesn't seem like the right place to be calling `appShutdown`. The emit below is supposed to trigger `BitcoinApplication::handleRunawayException`:

https://github.com/bitcoin/bitcoin/blob/59ecacfc84af13e5a1608e7d970315d07dcb0269/src/qt/bitcoin.cpp#L377

which is supposed to show an error dialog and then exit:

https://github.com/bitcoin/bitcoin/blob/59ecacfc84af13e5a1608e7d970315d07dcb0269/src/qt/bitcoin.cpp#L523-L527

If you shut down in `BitcoinCore::handleRunawayException`, it seems like this is going to delay the dialog from showing up, and maybe even prevent it from being seen by the user. But if this is really an improvement, it would be good to have a comment explaining what the shutdown sequence is.